lib/fog/google/storage.rb

```diff
@@ -1,292 +1,14 @@
-          @google_storage_access_key_id = options[:google_storage_access_key_id]
-          @google_storage_secret_access_key = options[:google_storage_secret_access_key]
-          @connection_options = options[:connection_options] || {}
-          @hmac = Fog::HMAC.new("sha1", @google_storage_secret_access_key)
-          @host = options[:host] || "storage.googleapis.com"
-          @persistent = options.fetch(:persistent, true)
-          @port       = options[:port] || 443
-          @scheme     = options[:scheme] || "https"
-          @path_style = options[:path_style] || false
-        end
-
-        def reload
-          @connection.reset if @connection
-        end
-
-        def signature(params)
-          string_to_sign =
-<<-DATA
-#{params[:method]}
-#{params[:headers]['Content-MD5']}
-#{params[:headers]['Content-Type']}
-#{params[:headers]['Date']}
-DATA
-
-          google_headers = {}
-          canonical_google_headers = ""
-          for key, value in params[:headers]
-            google_headers[key] = value if key[0..6] == "x-goog-"
-          end
-
-          google_headers = google_headers.sort { |x, y| x[0] <=> y[0] }
-          for key, value in google_headers
-            canonical_google_headers << "#{key}:#{value}\n"
-          end
-          string_to_sign << "#{canonical_google_headers}"
-
-          canonical_resource = "/"
-          if subdomain = params.delete(:subdomain)
-            canonical_resource << "#{CGI.escape(subdomain).downcase}/"
-          end
-          canonical_resource << "#{params[:path]}"
-          canonical_resource << "?"
-          for key in (params[:query] || {}).keys
-            if %w(acl cors location logging requestPayment torrent versions versioning).include?(key)
-              canonical_resource << "#{key}&"
-            end
-          end
-          canonical_resource.chop!
-          string_to_sign << "#{canonical_resource}"
-
-          signed_string = @hmac.sign(string_to_sign)
-          Base64.encode64(signed_string).chomp!
-        end
-
-        def connection(scheme, host, port)
-          uri = "#{scheme}://#{host}:#{port}"
-          if @persistent
-            unless uri == @connection_uri
-              @connection_uri = uri
-              reload
-              @connection = nil
-            end
-          else
-            @connection = nil
-          end
-          @connection ||= Fog::XML::Connection.new(uri, @persistent, @connection_options)
-        end
-
-        private
-
-        def request(params, &block)
-          params = request_params(params)
-          scheme = params.delete(:scheme)
-          host   = params.delete(:host)
-          port   = params.delete(:port)
-
-          params[:headers]["Date"] = Fog::Time.now.to_date_header
-          params[:headers]["Authorization"] = "GOOG1 #{@google_storage_access_key_id}:#{signature(params)}"
-
-          connection(scheme, host, port).request(params, &block)
+      def self.new(options)
```

Comment:
@icco @deanputney - This now breaks scenarios when Fog::Storage::Google.new is invoked without arguments, for example when `~/.fog` file is used.
Is this by design?

This happens because the old storage connection was invoked with an options hash , while now it takes a required argument.

For repro just run the storage example:
```
ruby storage_json.rb                                                                                                                                          (1)
/Users/temikus/Code/fog-google/lib/fog/google/storage.rb:7:in `new': wrong number of arguments (0 for 1) (ArgumentError)
	from storage_json.rb:14:in `test'
	from storage_json.rb:37:in `<main>'"
```